lego/apps/polls/serializers.py

```diff
@@ -0,0 +1,148 @@
+from django.db import transaction
+from rest_framework import serializers
+from rest_framework.fields import CharField, IntegerField
+
+from lego.apps.comments.serializers import CommentSerializer
+from lego.apps.polls.models import Option, Poll
+from lego.apps.tags.serializers import TagSerializerMixin
+from lego.utils.serializers import BasisModelSerializer
+
+
+class OptionSerializer(BasisModelSerializer):
+    class Meta:
+        model = Option
+        fields = ("id", "name", "votes")
+
+
+class OptionUpdateSerializer(OptionSerializer):
+
+    id = IntegerField(read_only=False)
+
+
+class PollSerializer(BasisModelSerializer):
+
+    comments = CommentSerializer(read_only=True, many=True)
+    comment_target = CharField(read_only=True)
+
+    options = OptionSerializer(many=True)
+    total_votes = IntegerField(read_only=True)
+
+    has_answered = serializers.SerializerMethodField()
+
+    def get_has_answered(self, obj):
+        user = self.context["request"].user
+        return user in obj.answered_users.all()
+
+    class Meta:
+        model = Poll
+        fields = (
+            "id",
+            "created_at",
+            "title",
+            "description",
+            "options",
+            "total_votes",
+            "tags",
+            "comments",
+            "comment_target",
+            "has_answered",
+            "pinned",
+        )
+
+
+class DetailedPollSerializer(TagSerializerMixin, BasisModelSerializer):
+
+    comments = CommentSerializer(read_only=True, many=True)
+    comment_target = CharField(read_only=True)
+
+    options = OptionSerializer(many=True)
+    total_votes = IntegerField(read_only=True)
+
+    has_answered = serializers.SerializerMethodField()
+
+    def get_has_answered(self, obj):
+        user = self.context["request"].user
+        return user in obj.answered_users.all()
```

Comment:
Will this retrieve all users from db to compare in python? A better solution would be to use a proper query to check.

Reply:
This will also generate a query per poll, so when listing polls, this will be quite ineffective. You can instead aggregate the data when selecting the queryset